#KPR-F4 first coderewiev. without tests (Get all active keepers)

dmitriyroy opened the pull request:
Первый пулреквест. На первый просмотр. Тестов еще нет.

dmitriyroy committed:
наличия develop

dmitriyroy committed:
проверка наличия правок. все есть

dmitriyroy committed:
создал контроллер, бин, сервис и репо Кипера

dmitriyroy committed:
создал RestKeeperReository, который имплементит KeeperRepository

dmitriyroy committed:
выпушил для проверки Вадимом

dmitriyroy committed:
внес правки Вадима, добавил кончтрукторы

dmitriyroy committed:
внес правки Вадима, добавил кончтрукторы

dmitriyroy committed:
добавил метку todo

dmitriyroy committed:
первый PullRequest на первый ревью кода

dmitriyroy committed:
#KPR-F4 первый пуллреквест на первый кодревью. тестоами еще не покрыто.

NikolayNN commented:
на будущее
Все комиты должны начинаться с номера таски, в рамках которой они созданы. и стараться их писать на английском.

dmitriyroy committed:
Merge remote-tracking branch 'upstream/develop' into develop

# Conflicts:
#	src/main/resources/application.properties

dmitriyroy committed:
#KPR-F4 починил проперти

dmitriyroy committed:
#KPR-F4 updated some classes after upstream fork

dmitriyroy committed:
#KPR-F4 add field isActive into entity Keeper
update select from keeper

OlgaKulikova commented:
@dmitriyroy В этот раз не меняй уже, но со следующей задачи создавай из своего develop отдельный бранч для фичи - по номеру таски. Это также описано в [Workflow](https://github.com/JujaLabs/docs/tree/master/other/workflow)

OlgaKulikova commented on keepers.iml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
oops, кто-то забыл добавить в гитигнор файлы *.iml
Эти файлы генерит Идея и они индивидуальны для каждого разработчика. В гите они не нужны и даже вредны. Нужно из гита его удалить (но не физически) и добавить в гитигнор *.iml

dmitriyroy replied:
Возможно это я. Добавил в игнор *.iml

OlgaKulikova commented on src/main/java/juja/microservices/keepers/controller/KeepersController.java:
```diff
@@ -41,6 +42,6 @@
     public ResponseEntity<?> getKeepers() {
         //TODO Should be implemented feature KPR-F4
-        return null;
+        return ResponseEntity.ok(keepersService.getAllActiveKeepers());
```
Если задача имплементирована - убираем Todo

+ просмотрел - вроде все свои тудушки убрал

adaleorion replied:
Нет логирования результата
тут нужно два уровня первый инфо, который покажет что все отработало успешно и запишет кол-вл найденных киперов,

второй на дебаг, в это случае выведем полное содержание полученное из сервиса

dmitriyroy replied:
по логированию - я еще не прикручивал вообще

OlgaKulikova replied:
Давайте воспринимать логирование как неотъемлемую часть работы над фичей (так же, как и тесты). @dmitriyroy добавь логи, это не сложно, конфигурация по нему уже есть. Если есть вопросы - как где и когда - спрашивай.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -1,9 +1,15 @@
+import java.util.*;
 
 /**
  * @author Vadim Dyachenko
```
также можно (и нужно) добавлять еще авторов, кто работает над данным классом

dmitriyroy replied:
добил себя соавтором с указанием фичи, по которой соавторствовал

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir33", new Date(),  null, true));
+
+        List<Keeper> keepers = mongoTemplate.find(new Query(Criteria.where("isActive").is(true)), Keeper.class,"keeper");
+        for (Keeper k:keepers) {
```
немного код конвеншена - пробелы до и после двоеточия, лучше не "k", а "keeper"

dmitriyroy replied:
готово - пробелы + keeper вместо k

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+
+        List<Keeper> keepers = mongoTemplate.find(new Query(Criteria.where("isActive").is(true)), Keeper.class,"keeper");
+        for (Keeper k:keepers) {
+            List<String> tmpList = new ArrayList<>();
```
также лучше назвать лист по содержимому, например, directions.

dmitriyroy replied:
сделал

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+    public Map<String, List<String>> getAllActiveKeepers(){
+        String urlTemplate = urlBase + urlGetKeepers;
+        Map<String, List<String>> outMap = new HashMap<>();
+//        mongoTemplate.insert(new Keeper("111", "den","dir11", new Date(),  new Date(), false));
```
Насколько я понимаю, это данные для первого заполнения, чтобы было что потом вытягивать. Если есть необходимость, можно написать скрипт для тестого наполнения базы, но такое точно не надо даже в комментах оставлять в коде. Закомментированный код также не приветствуется.

dmitriyroy replied:
убрал комменты все.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir23", new Date(),  null, true));
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir33", new Date(),  null, true));
+
+        List<Keeper> keepers = mongoTemplate.find(new Query(Criteria.where("isActive").is(true)), Keeper.class,"keeper");
```
Для первого раза пойдет. Но в дальнейшем надо подумать на счет агрегированного запроса в монго, чтобы не нужно было руками собирать мапу.

adaleorion replied:
поддерживаю думаю нужно с map поиграться и вместо нее использовать сущность кипер + направления получиться в dto, который можно будет спокойно вычитать из базы

dmitriyroy replied:
пока над мапой не игрался.

OlgaKulikova replied:
можно здесь пока оставить как есть, но добавить туду в коде и создать задачу "Optimize Query to DB with result aggregation" (или как-то так) В самой задаче можно объяснить подробней, что есть сейчас, что хотелось бы видеть.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/exception/ApiErrorMessage.java:
```diff
@@ -50,7 +50,11 @@ private ApiErrorMessage(String code, String clientMessage, String developerMessa
     }
 
-     static class ApiErrorMessageBuilder {
+    public String getExceptionMessage() {
```
геттер должен работать с аннотации, если все правильно настроено в Идее

dmitriyroy replied:
убрал геттеры. подключил lombok. вроде заработало с ним.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/service/KeepersService.java:
```diff
@@ -13,4 +16,8 @@
     @Inject
     private KeepersRepository keepersRepository;
     //TODO Should be implemented internal service logic
```
не забываем убирать тудушки

dmitriyroy replied:
прошелся по коду - поудалял сделанные свои туду

OlgaKulikova commented on src/main/java/juja/microservices/keepers/service/KeepersService.java:
```diff
@@ -1,9 +1,12 @@
 package juja.microservices.keepers.service;
 
 import juja.microservices.keepers.dao.KeepersRepository;
+import juja.microservices.keepers.exceptions.UserMicroserviceExchangeException;
```
а зачем здесь этот импорт?

dmitriyroy replied:
удалил ненужные импорты. помнится, idea сама удаляла неюзаемые импорты, поэтому и не проверял.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/exceptions/UserMicroserviceExchangeException.java:
```diff
@@ -0,0 +1,6 @@
+package juja.microservices.keepers.exceptions;
+
+public class UserMicroserviceExchangeException extends Throwable {
+    public UserMicroserviceExchangeException(String s) {
```
а где применяется этот эксепшн?

dmitriyroy replied:
этот эксепшин был сделан Вадимом во время заготовки всего макета сервиса. не удаляю пока его.

OlgaKulikova replied:
странно, что он был сделан раньше, а у тебя он выглядит как новый код.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
 
     //TODO Should be implemented work with MongoDB
+
+    @Value("${user.baseURL}")
```
а почему user? думаю, здесь должен быть keepers (соответственно и в application.properties)

adaleorion replied:
зачем нам url, или будем ходить по rest на удаленные сервисы ?

dmitriyroy replied:
убрал вообще эти урлы и из проперти тоже

OlgaKulikova reviewed (commented):
Please look at the commets and do changes.

adaleorion commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir13", new Date(),  null, true));
+//        mongoTemplate.insert(new Keeper("fedor", "vadim","dir22", new Date(),  null, true));
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir23", new Date(),  null, true));
+//        mongoTemplate.insert(new Keeper("sollyk", "sanek","dir33", new Date(),  null, true));
```
для тестирования лучше использовать fakemongo и загонять туда тестовые наборы данных

+ еще по аналогии с другими методами нужно добавить логирование на debug

dmitriyroy replied:
возьму на заметку. пока просто удалил этот блок, т.к. базу уже тестово заполнил

adaleorion commented on src/main/java/juja/microservices/keepers/dao/KeepersRepository.java:
```diff
@@ -15,4 +21,38 @@
+    private String urlGetKeepers;
+
+    public Map<String, List<String>> getAllActiveKeepers(){
+        String urlTemplate = urlBase + urlGetKeepers;
```
зачем нам url, или будем ходить по rest на удаленные сервисы ?

dmitriyroy replied:
логирование не прикручивал.
урлы поудалял в том числе и из проперти

adaleorion commented on src/main/java/juja/microservices/keepers/entity/Keeper.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    @Override
+    public String toString() {
```
toString переорпделять не нужно, ты же указал аннотации ломбока, нужно тогда или аннотации убрать или методы. А так полное дублирование получилось.
Тоже самое и с конструкторами

dmitriyroy replied:
с мпаой не игрался еще.
toString() удалил

dmitriyroy committed:
#KPR-F4 add fixed after codereview
- delete //todo
- delete URL from properties
- add code convention
- delete comments

OlgaKulikova commented:
> Читаю Workflow уже раз пятнадцатый
Там написано: "Формат названия ветки "task number"-"feature name". Пример: 123-important-feature."
((( - что за номер таски? Названии фичи - все понятно - KPR-F4, а вот номер таски где взять, не ясно.

Номер таски - это номер issue на Гитхаб, когда ты берешься делать задачу. Одна таска (issue)  = один пул реквест.

OlgaKulikova commented on src/main/java/juja/microservices/keepers/Keepers.java:
```diff
@@ -21,6 +21,7 @@
 
 /**
  * @author Vadim Dyachenko
+ * KPR-F4 Dmitriy Roy
```
ну не знаю, надо ли добавлять номер фичи. Я предполагала просто еще одну строку `@author Dmitriy Roy`

OlgaKulikova commented on src/main/java/juja/microservices/keepers/entity/Keeper.java:
```diff
@@ -27,45 +30,33 @@
 
-    @JsonCreator
-    public Keeper() {
+//    @JsonCreator
```
ойой - закомментированный код! убираем.

dmitriyroy committed:
#KPR-F4 delete // comments

dmitriyroy committed:
#KPR-F4 rename exception

dmitriyroy committed:
#KPR-F4 delete future number from block - javadoc

OlgaKulikova commented on src/main/java/juja/microservices/keepers/Keepers.java:
```diff
@@ -21,7 +21,7 @@
 /**
  * @author Vadim Dyachenko
- * KPR-F4 Dmitriy Roy
+ *         Dmitriy Roy
```
и все-таки я прошу ставить еще одну строчку с собакой и словом author перед твоей фамилией. Джава доки вытягиваются автоматом в документ, ориентируясь на определенные символы. Без собаки и слова author оно не распознает, что ты автор.

dmitriyroy replied:
готово

dmitriyroy committed:
#KPR-F4 @author into block - javadoc
